thorlcr/graph/thgraphmaster.cpp

```diff
@@ -649,25 +649,48 @@ void CMasterActivity::done()
 
 void CMasterActivity::updateFileReadCostStats()
 {
-    // Updates numDiskReads & readCost in the file attributes and returns the readCost
-    auto updateReadCosts = [](IDistributedFile *file, CThorStatsCollection &stats)
+    // Update numDiskReads & readCost in the file attributes and return readCost
+    auto updateReadCosts = [](bool useJhtreeCacheStats, IDistributedFile *file, CThorStatsCollection &stats)
     {
-        stat_type curDiskReads = stats.getStatisticSum(StNumDiskReads);
+        StringBuffer clusterName;
+        file->getClusterName(0, clusterName);
         IPropertyTree & fileAttr = file->queryAttributes();
-        cost_type legacyReadCost = getLegacyReadCost(fileAttr, file);
-        cost_type curReadCost = money2cost_type(calcFileAccessCost(file, 0, curDiskReads));
+        cost_type legacyReadCost = 0, curReadCost = 0;
+        // Legacy files will not have the readCost stored as an attribute
+        if (!hasReadWriteCostFields(fileAttr) && fileAttr.hasProp(getDFUQResultFieldName(DFUQRFnumDiskReads)))
+        {
+            // Legacy file: calculate readCost using prev disk reads and new disk reads
+            stat_type prevDiskReads = fileAttr.getPropInt64(getDFUQResultFieldName(DFUQRFnumDiskReads), 0);
+            legacyReadCost = money2cost_type(calcFileAccessCost(clusterName, 0, prevDiskReads));
+        }
+        stat_type curDiskReads = stats.getStatisticSum(StNumDiskReads);
+        if(useJhtreeCacheStats)
+        {
+            stat_type numActualReads = stats.getStatisticSum(StNumNodeDiskFetches)
```

Comment:
Should this code be setting curDiskReads?  It will affect the number of reads recorded against the file in line 675.  (It would also allow the cost calculation to be shared.)

Reply:
I recall @jakesmith mentioned that it would be useful to have both: the original NumDiskReads and the jhtree stats.

Reply:
I think it is fine as it stands.
As it is, we have 2 pieces of info., we have the number of attempts to read from disk (StNumDiskReads) and the number of actual disk reads, at least in the index case (StNumNodeDiskFetches+StNumLeafDiskFetches+StNumBlobDiskFetches).

StNumDiskReads is a tad misleading (and always was), and may not closely correspond to actual disk reads in flat files in some cases either. 
The explanation for what it is (and other stats) is something that should be clarified somewhere, but not directly related to this PR I would say.